prow/config/config.go

```diff
@@ -913,27 +913,36 @@ func setPeriodicDecorationDefaults(c *Config, ps *Periodic) {
 // finalizeJobConfig mutates and fixes entries for jobspecs
 func (c *Config) finalizeJobConfig() error {
 	if c.decorationRequested() {
-		if c.Plank.DefaultDecorationConfig == nil {
-			return errors.New("no default decoration config provided for plank")
-		}
-		if c.Plank.DefaultDecorationConfig.UtilityImages == nil {
-			return errors.New("no default decoration image pull specs provided for plank")
-		}
-		if c.Plank.DefaultDecorationConfig.GCSConfiguration == nil {
-			return errors.New("no default GCS decoration config provided for plank")
-		}
-		if c.Plank.DefaultDecorationConfig.GCSCredentialsSecret == "" {
-			return errors.New("no default GCS credentials secret provided for plank")
-		}
+		if len(c.Plank.DefaultDecorationConfigs) > 0 {
+			_, ok := c.Plank.DefaultDecorationConfigs["*"]
```

Comment:
This is still more complicated than needed... Let me try to give some pseudocode:

```
if config != nil:
  if configs != nil:
    return Error("can't have both, please have just configs as its new and cool")
  log.Warn("you are using old config, we will deprecate it in 2020, use new config")
  configs := configs{"*": config}
if configs == nil:
  return Error("you do not have any default config") # no need to mention old config here - direct users to new if they have none
if configs["*"] == nil:
  return Error("you do not have "*"")
if !configs["*"].Validate():
  return Error("invalid config")
```

Unless I missed something, this should cover everything that is needed. Compared to the PR, it does not allow having old config together with new config without "*", but that is IMO reasonable anyway - if someone wants new feature, let him migrate fully.